xbmc/video/VideoThumbLoader.cpp

```diff
@@ -451,21 +453,39 @@ bool CVideoThumbLoader::FillLibraryArt(CFileItem &item)
       if (database.GetArtForItem(idAlbum, MediaTypeAlbum, artwork))
         item.SetArt(artwork);
     }
-    // For episodes and seasons, we want to set fanart for that of the show
-    if (!item.HasArt("fanart") && tag.m_iIdShow >= 0)
+
+    if (tag.m_type == MediaTypeEpisode || tag.m_type == MediaTypeSeason)
     {
-      ArtCache::const_iterator i = m_showArt.find(tag.m_iIdShow);
-      if (i == m_showArt.end())
+      // For episodes and seasons, we want to set fanart for that of the show
+      if (!item.HasArt("fanart") && tag.m_iIdShow >= 0)
       {
-        std::map<std::string, std::string> showArt;
-        m_videoDatabase->GetArtForItem(tag.m_iIdShow, MediaTypeTvShow, showArt);
-        i = m_showArt.insert(std::make_pair(tag.m_iIdShow, showArt)).first;
+        ArtCache::const_iterator i = m_showArt.find(tag.m_iIdShow);
+        if (i == m_showArt.end())
+        {
+          std::map<std::string, std::string> showArt;
+          m_videoDatabase->GetArtForItem(tag.m_iIdShow, MediaTypeTvShow, showArt);
+          i = m_showArt.insert(std::make_pair(tag.m_iIdShow, showArt)).first;
+        }
+        if (i != m_showArt.end())
+        {
+          item.AppendArt(i->second, "tvshow");
+          item.SetArtFallback("fanart", "tvshow.fanart");
+          item.SetArtFallback("tvshow.thumb", "tvshow.poster");
+        }
       }
-      if (i != m_showArt.end())
+
+      if (!item.HasArt("season.poster") && tag.m_iSeason > 0)
```

Comment:
Wouldn't this exclude the logic for the special seasons? I thought they are listed as Season 0?